src/SFA.DAS.EmployerAccounts/Queries/GetAccountPayeSchemes/GetAccountPayeSchemesQueryHandler.cs

```diff
@@ -41,24 +36,11 @@ public async Task<GetAccountPayeSchemesResponse> Handle(GetAccountPayeSchemesQue
             {
                 throw new UnauthorizedAccessException();
```

Comment:
Isn't this code now redundant as the validator no longer checks authorisation?

Reply:
It is if I remove the test that covers it as well, which I have done.

However I have a bit of an issue with this; not with your comment, but in the whole approach.

So I could argue that the line should stay in, because the Handler doesn't know whether the Validator might return unauthorized or not, that is not a 'concern' of the Handler. A `ValidationResult` could have `IsUnauthorized` set.

Thinking about it my issue is more that we have this same code in all the handlers. It seems like the validation should be handled outside of the handlers or maybe the validators should be throwing their own exception.